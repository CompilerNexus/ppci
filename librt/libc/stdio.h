#ifndef STDIO_H
#define STDIO_H

void printf(char*);
void bsp_putc(char);

#endif

