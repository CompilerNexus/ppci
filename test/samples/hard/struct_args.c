
#include <stdio.h>

typedef struct { int a; char b; int c; } foo_t;

void do3(int x, foo_t y, int z)
{
  printf("Enter do\n");
  int g;
  g = x + y.a + (int)y.b + y.c + z;
  printf("v:%d\n", g);
  printf("Exit do\n");
}

void main_main()
{
  printf("Enter main\n");
  foo_t bar;
  bar.a = 5;
  bar.b = 99;
  bar.c = 124;
  do3(0x1337, bar, 0x42);
  printf("Exit main\n");
}

