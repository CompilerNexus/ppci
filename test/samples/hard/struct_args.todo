
#include <stdio.h>

typedef struct { int a; char b; int c; } foo_t;

void do3(int x, foo_t y, int z)
{
  printf("Enter do\n");
  volatile int g = x + y.a + y.b + y.c + z;
  // # print(g)
  printf("Exit do\n");
}

void main_main()
{
  printf("Enter main\n");
  foo_t bar;
  bar.a = 5;
  bar.b = 99;
  bar.c = 124;
  do3(0x1337, bar, 0x42);
  printf("Exit main\n");
}

