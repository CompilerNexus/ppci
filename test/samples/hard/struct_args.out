Enter main
Enter do
v: 0000145D
Exit do
Exit main
